pandas/core/internals.py

```diff
@@ -2621,6 +2598,13 @@ def __init__(self, values, placement, ndim=2, **kwargs):
         super(DatetimeTZBlock, self).__init__(values, placement=placement,
                                               ndim=ndim, **kwargs)
 
+    def get_values(self, dtype=None):
+        """
+        return object dtype as boxed values, such as Timestamps/Timedelta
+        """
+        # Note: this overrides the NonConsolidatableMixIn implementation
+        return DatetimeBlock.get_values(self, dtype)
```

Comment:
so rather than do this, move the impl from DatetimeBlock here, you are calling a sub-class, not a good idea

Reply:
You want to copy/paste the method?  The original version in the PR was just `get_values = DatetimeBlock.get_values` to override the version from NonConsolidatableMixin (which is higher in the MRO).

Reply:
IIUC, the idea was to

1. Move `DatetimeBlock.get_values` here to the mixin
2. Define `DatetimeTZBlock.get_values` as

```python
def get_values(self, dtype=None):
    return DatetimeLikeBlockMixin.get_values(self, dtype)
```

Or, does changing DatetimeTZBlock's MRO to `(DatetimeLikeBlockMixin, NonConsolidatableMixin, DatetimeBlock) break anything?

Reply:
`DatetimeLikeBlockMixin` already defines get_values.  The original version from this PR `get_values = DatetimeBlock.get_values` could equivalently be `get_values = DatetimeLikeBlockMixin.get_values`, which I guess is slightly better.

> Or, does changing DatetimeTZBlock's MRO to (DatetimeLikeBlockMixin, NonConsolidatableMixin, DatetimeBlock) break anything?

Haven't tried that, but I expect it would break a lot b/c I think _nothing_ from `NonConsolidatableMixin` would get used.

Reply:
Ah that wouldn't work since you'd be inheriting from `DatetimeLikeBlockMixin` twice.

Reply:
If this is just calling the super method what happens if you just remove it entirely?

Reply:
> If this is just calling the super method what happens if you just remove it entirely?

 Argh.  If this is removed entirely then the `NonConsolidatableMixin.get_values` implementation is used because it is at the top of the MRO.  (I haven't checked how it breaks, but am assuming that whoever wrote the existing method overrode it for a reason).  That's why all that is needed is a one-liner to point it towards the DatetimeBlock.get_values method.